app/controllers/admin/api/cms/templates_controller.rb

```diff
@@ -2,12 +2,19 @@ class Admin::Api::CMS::TemplatesController < Admin::Api::CMS::BaseController
   ##~ sapi = source2swagger.namespace("CMS API")
   ##~ @parameter_template_id = { :name => "id", :description => "ID of the template", :dataType => "int", :required => true, :paramType => "path" }
 
-  ALLOWED_PARAMS = %i[type system_name title path draft liquid_enabled handler content_type section_id section_name layout_id layout_name].freeze
+  AVAILABLE_PARAMS = %i[system_name title path draft liquid_enabled handler content_type section_id layout_id].freeze
```

Comment:
With this change, the possibility to update the templates' layout and section using the system names (`layout_name`, `section_name`) is removed. Is that deliberate?

Reply:
Yes, this list is used to filter permitted params for mass-assignment, which must have the same name as DB columns. `layout_name` and `section_name` are still accepted but are not permitted for mass-assignment, `TemplateService` takes care of them manually.